azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -461,6 +471,11 @@ public int selectAndUpdateExecutionWithLocking(final int executorId, final boole
         "SELECT exec_id, enc_type, flow_data, status FROM execution_flows "
             + "WHERE project_id=? AND flow_id=? AND status=? "
             + "ORDER BY exec_id DESC LIMIT ?, ?";
+    // Fetch flows that are in preparing state for more than a certain duration.
+    private static final String FETCH_FLOWS_QUEUED_FOR_LONG_TIME =
+        "SELECT exec_id, enc_type, flow_data, status FROM execution_flows"
+            + " WHERE submit_time < ? AND status = "
+            + Status.PREPARING.getNumVal();
```

Comment:
Is it possible to return the number of executions matching the criteria only? Since this query is executed every minute and it's only used for monitoring purposes it would be good to avoid the overhead of converting the `flow_data` blob into a Java object.

Reply:
Originally, I started in the direction suggested by you. It involved creating a different class parallel to FetchExecutableFlows. This new class was a 3rd class for operations that were very similar. I was asked to merge the class into the existing one. 

The existing classes, one of which I am reusing, were doing redundant work when it came to computing number of flows:
- "WEB-NumQueuedFlows",  is using this.executorManagerAdapter::getQueuedFlowSize.
- "WEB-NumRunningFlows", is using this.executorManagerAdapter.getRunningFlows().size()

(These existing gauges, in fact, have an added redundant burden of pairing up with ExecutionReference as well.)

I then thought I will reuse existing class for now, especially because this wasn't having any material impact on the performance of Web server's core operations, and then follow-up to pull out NumQueuedFlow and NumQueuedFlows.  Let's do this as a follow-on.

Reply:
Ok. Could you add a TODO comment summarizing what's pending?

Reply:
Done